pallene/to_ir.lua

```diff
@@ -89,52 +89,56 @@ function ToIR:convert_stat(cmds, stat)
             ir.Cmd.Seq(body)))
 
     elseif tag == "ast.Stat.Assign" then
-        local var = stat.var
-        local exp = stat.exp
-        if     var._tag == "ast.Var.Name" then
-            local cname = var._name
-            if     cname._tag == "checker.Name.Local" then
-                self:exp_to_assignment(cmds, cname.id, exp)
-            elseif cname._tag == "checker.Name.Global" then
-                local v = self:exp_to_value(cmds, exp)
-                table.insert(cmds, ir.Cmd.SetGlobal(stat.loc, cname.id, v))
-            else
-                error("impossible")
-            end
-
-        elseif var._tag == "ast.Var.Bracket" then
-            local arr = self:exp_to_value(cmds, var.t)
-            local i   = self:exp_to_value(cmds, var.k)
-            local v   = self:exp_to_value(cmds, exp)
-            local src_typ = exp._type
-            table.insert(cmds, ir.Cmd.SetArr(stat.loc, src_typ, arr, i, v))
+        for i = 1, #stat.vars do
```

Comment:
We should be careful to keep the order of evaluation the same as Lua. https://www.lua.org/manual/5.3/manual.html#3.3.3

Reply:
It should be working the same as Lua. I wrote some tests to verify that too, with similar examples to the ones in Lua's manual.